lib/src/authorization_code_grant.dart

```diff
@@ -175,13 +187,20 @@ class AuthorizationCodeGrant {
       scopes = scopes.toList();
     }
 
+    _codeVerifier = _createCodeVerifier();
+    var codeChallenge = base64Url
+        .encode(sha256.convert(ascii.encode(_codeVerifier)).bytes)
```

Comment:
Are you sure this is `ascii` and not `utf8` ? If so should we check and throw if the string is not ASCII?

Reply:
According to [RFC7636](https://tools.ietf.org/html/rfc7636#section-4.2), it should be ASCII.

> The client then creates a code challenge derived from the code verifier by using one of the following transformations on the code verifier:
> plain
      code_challenge = code_verifier
> S256
      code_challenge = BASE64URL-ENCODE(SHA256(ASCII(code_verifier)))

No need to check if the string is ASCII. The ASCII encode will work in both cases